walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -15,9 +15,96 @@
+    HashMap<Integer, ArrayList<Integer>> mandatoryEvents = new HashMap<Integer, ArrayList<Integer>>();
+
+    for (Event event : allEvents) {
+      //people who must attend 
+      Set<String> people = event.getAttendees();
+      //check if any people in meeting request is present in this set
+      for (String person : attendees) {
+        if (people.contains(person)) {
+          // If event already exists in hash map, add the end time to the ArrayList of end times. Otherwise, create an ArrayList for end times and add event to hash map.
+          int eventStartTime = event.getWhen().start();
+          int eventEndTime = event.getWhen().end();
+          if (mandatoryEvents.containsKey(eventStartTime)) {
+            ArrayList<Integer> currEndTimes = mandatoryEvents.get(eventStartTime);
+            currEndTimes.add(eventEndTime);
+          }
+          else {
+            ArrayList<Integer> newEndTimes = new ArrayList<Integer>();
+            newEndTimes.add(eventEndTime);
+            mandatoryEvents.put(eventStartTime, newEndTimes);
+          }
+        }  
+      }
+    }
+
+    int currTime = TimeRange.START_OF_DAY;
+    int startTime = TimeRange.START_OF_DAY;
+    int currDuration = 0;
+
+    while (currTime < TimeRange.END_OF_DAY) {
+      /* If there's an event starting at this time, compare the current duration and meeting duration to see if there is a possible meeting time. 
+        Reset the current duration variable, move the current time to the end time of the event, and repeat loop. */
+      if (mandatoryEvents.containsKey(currTime)) {
+        ArrayList<Integer> endTimes = mandatoryEvents.get(currTime);
+        // Get list item with latest end time
+        int longestMeetingEnd = Collections.max(endTimes); 
+        if (currDuration >= mtngDuration) {
+            // Not inclusive because a meeting starts at this time 
+            meetingTimes.add(TimeRange.fromStartEnd(startTime, currTime, false));
+        }
+        currDuration = 0;
+
+        // Check if there is a conflict: another meeting that starts during this one, and goes later
+        int tempTime = currTime + 10;
+        int currentEventEnd = longestMeetingEnd;
+        while (tempTime < currentEventEnd) {
+          if (mandatoryEvents.containsKey(tempTime)) {
+            ArrayList<Integer> conflictMeetingEndTimes = mandatoryEvents.get(tempTime);
+            int meetingEnd = Collections.max(conflictMeetingEndTimes);
+            if (meetingEnd > longestMeetingEnd) {
+              longestMeetingEnd = meetingEnd;
+            }
+          }
+          tempTime += 10;
+        }
+          
+      currTime = longestMeetingEnd;
+      startTime = longestMeetingEnd;
+      }
+
+      /* If there is no event at this time, add 30 minutes to the duration and current time, and run the loop again. */
+      else {
+        currDuration += 10;
+        currTime +=10;
+       }
+    }
```

Comment:
This is looking good, but it's pretty long for one method so I wonder if you can break this up into more logical helper methods.  From the looks of it, some of the steps you're doing are:
1) setting up the map of event times w/ conflicts
2) processing the map to determine where the "free time" is in the map 

Maybe at least those can be broken up?  Smaller helper methods will also help document the code.